DEVELOPER_GUIDE.md

```diff
@@ -53,10 +54,43 @@ Bound addresses will then be logged to the terminal :
 [main] INFO  transportservice.TransportService - profile [test]: publish_address {127.0.0.1:5555}, bound_addresses {[::1]:5555}, {127.0.0.1:5555}
 ```
 
+
+
+## Create extensions.yml file
+
+Every extension will require metadata stored in an extensions.yml file in order to be loaded successfully.  In order to make the SDK look like an extension within OpenSearch, there must be an entry for the SDK within `extensions.yml`.
+
+- Start a separate terminal and navigate to the directory that OpenSearch has been cloned to using `cd OpenSearch`.
+- Run `./gradlew assemble` to create a local distribution.
+- Navigate to the project root directory (i.e. `cd distribution/archives/linux-tar/build/install/opensearch-3.0.0-SNAPSHOT/`). Note: On Mac OS `linux-tar` should be replaced with `darwin-tar`.
+
+- Check if extensions directory exists in OpenSearch using `ls`.
```

Comment:
(Future suggestion, don't hold up PR review for this) All these steps are just screaming for automation.  We could replace all these steps with `./createBootstrapExtensionsYaml.sh` including writing the sample file.